### DISCUSSION TITLE
MCycle.v is for ARM ISA

### DISCUSSION THREAD

**Discussion by tayponghee** [NUS-CG3207/labs#26] [Lab 3]
In the MCycle.v file, the inputs and outputs are meant for ARM processors, and not Risc-V ones.
![image](https://github.com/user-attachments/assets/f938a0ab-3909-4dc3-886b-88e0eb082b42)

May I check if there is an updated file that supports Risc-V? Thank you.

**Comment by r-panicker** [comment]
It is ISA independent*, though originally written for ARM.
Thanks for pointing out, we will remove the references to ARM.
*mulhsu support requires MCycleOp to be more than 2 bits though, as it is not covered by the 4 possibilities - signed and unsigned multiplication and division.
